build.sbt

```diff
@@ -117,6 +117,29 @@ lazy val root = (project in file("."))
     )
   )
 
+lazy val scalaVersionDependentSettings = Def.settings(
+  libraryDependencies ++= (if (scalaVersion.value.startsWith("2."))
+                             COMPILE(
+                               scalaVersion("org.scala-lang" % "scala-reflect" % _).value
+                             )
+                           else Seq.empty),
+  scalacOptions ++= (if (scalaVersion.value.startsWith("2."))
+                       Seq(
+                         "-Ywarn-macros:after",
+
+                         // Ensure we publish an artifact linked to the appropriate Java std library.
+                         "-release:17",
```

Comment:
This flag MUST be always added, so let's move it outside of this block.
Actually, it is already part of the `commonSettings`, so there is no need to re-add it here.

Reply:
Addressed.